core/src/main/scala/akka/persistence/jdbc/schema/scaladsl/SchemaUtils.scala

```diff
@@ -0,0 +1,131 @@
+import slick.jdbc.SQLServerProfile
+
+object SchemaUtils {
+
+  private val logger = LoggerFactory.getLogger("akka.persistence.jdbc.schema.SchemaUtils")
+
+  /**
+   * For testing purposes the journal and snapshot tables can be created programmatically.
+   * For production it's recommended to create the table with DDL statements before the system is started.
+   * For information about the different schemas and supported database can be found at
+   * https://doc.akka.io/docs/akka-persistence-jdbc/current/index.html#database-schema
+   *
+   * This method will automatically detect the configured database using the settings from `jdbc-journal` config.
+   * If configured with `use-shared-db`, it will use the `akka-persistence-jdbc.shared-databases` definition instead.
+   * See https://doc.akka.io/docs/akka-persistence-jdbc/current/index.html#sharing-the-database-connection-pool-between-the-journals for details.
+   *
+   * @param systemProvider a ClassicActorSystemProvider
+   */
+  def dropAndCreate(implicit systemProvider: ClassicActorSystemProvider): Future[Done] = {
+
+    val actorSystem = systemProvider.classicSystem
+    val journalConfig = actorSystem.settings.config.getConfig("jdbc-journal")
+    val slickDb: SlickDatabase = SlickExtension(systemProvider).database(journalConfig)
+
+    val (file, separator) =
+      slickDb.profile match {
+        case PostgresProfile  => ("schema/postgres/postgres-schema.sql", ";")
+        case MySQLProfile     => ("schema/mysql/mysql-schema.sql", ";")
+        case OracleProfile    => ("schema/oracle/oracle-schema.sql", "/")
+        case SQLServerProfile => ("schema/sqlserver/sqlserver-schema.sql", ";")
+        case H2Profile        => ("schema/h2/h2-schema.sql", ";")
+      }
+
+    val blockingEC = actorSystem.dispatchers.lookup(Dispatchers.DefaultBlockingDispatcherId)
+    applySchema(slickDb.database, file, separator, blockingEC)
+  }
+
+  /**
+   * For testing purposes the journal and snapshot tables can be created programmatically.
+   *
+   * This method can be used to load alternative schema scripts. The passed schema argument must be a file accessible from the classpath.
+   *
+   * By default it will use the database settings found under `jdbc-journal`, or `akka-persistence-jdbc.shared-databases` if configured so.
+   * See https://doc.akka.io/docs/akka-persistence-jdbc/current/index.html#sharing-the-database-connection-pool-between-the-journals for details.
+   *
+   * To use the snapshot configuration, pass `jdbc-snapshot-store` to `configKey`.
+   *
+   * @param schema the DDL script file. Must be accessible from the classpath.
+   * @param separator used to separate the different DDL statement. Defaults to ';' for most databases.
+   * @param configKey the configuration key to use, default to `jdbc-journal`
+   * @param systemProvider a ClassicActorSystemProvider
+   * @return
+   */
+  def applySchema(schema: String, separator: String = ";", configKey: String = "jdbc-journal")(
```

Comment:
Default params are problematic for bincomp, not sure what we promise around that for akka-persistence-jdbc?

Reply:
We do, this should be done with an overload instead.

Reply:
I'd say that bincompat is a lost promise for this plugin until we rewrite it, but good to not introduce more warts